static/js/containers/PlotContainer.jsx

```diff
@@ -1,12 +1,70 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 
 import Plot from '../components/Plot';
-import { FETCH_SOURCE_PLOT } from '../actions';
-import * as API from '../API';
+import * as Actions from '../actions';
 
-const fetchPlotData = url => (
-  API.GET(url, FETCH_SOURCE_PLOT)
+
+class PlotContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: false,
+      fetchingPlotIDs: []
+    };
+  }
+  componentWillMount() {
+    this.fetchPlotDataIfNotCached();
+  }
+  componentWillReceiveProps() {
+    this.fetchPlotDataIfNotCached();
+  }
+  fetchPlotDataIfNotCached() {
+    if (!this.props.plots.plotIDList.includes(this.props.url) &&
```

Comment:
This if check may be better refactored into a function, like `needsFetching`, which can be reused below.  Actually, the part below can probably just go in an `else` clause?

Reply:
No, they're totally different checks. One is checking if the data needs to be fetched, and the other is checking if the fetch is complete and the ID needs to be removed from the list of in-process fetches (that is, it's complete).

Reply:
I see, you're right.